src/github_runner_manager/openstack_cloud/health_checks.py

```diff
@@ -0,0 +1,241 @@
+#  Copyright 2024 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Collection of functions related to health checks for a runner VM."""
+
+import logging
+from datetime import datetime, timedelta
+
+import invoke
+from fabric import Connection as SSHConnection
+
+from github_runner_manager.errors import KeyfileError, SSHError
+from github_runner_manager.manager.cloud_runner_manager import CloudInitStatus, CloudRunnerState
+from github_runner_manager.openstack_cloud.constants import (
+    METRICS_EXCHANGE_PATH,
+    RUNNER_LISTENER_PROCESS,
+    RUNNER_WORKER_PROCESS,
+)
+from github_runner_manager.openstack_cloud.openstack_cloud import OpenstackCloud, OpenstackInstance
+from github_runner_manager.utilities import retry
+
+logger = logging.getLogger(__name__)
+
+_HealthCheckResult = bool | None  # None indicates that the check can not determine health status
+
+
+def check_runner(openstack_cloud: OpenstackCloud, instance: OpenstackInstance) -> bool:
```

Comment:
Let's put a ticket on the backlog to improve this

Reply:
Done